### PR TITLE
IntakeReporter

### PR DESCRIPTION
**Pull request by pkarman**
### Description
Adds new class IntakeReporter and a small refactor for existing *Reporter classes to DRY them up.

### Documentation Updates
- [x] Topline File Descriptions Added or Updated as Needed

### CODE REVIEW

File: app/services/intake_reporter.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @start_date = start_date
+    @end_date = end_date
+    @type = type
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def report
+    establishment_durations = []
+    intakes.each do |intake|
+      establishment_durations << (intake.completed_at - intake.started_at).to_i
```
Comment (leikkisa) [inline app/services/intake_reporter.rb]: The calculation for establishment time would be `intake.detail.establishment_processed_at - intake.detail.establishment_submitted_at`.  The metric here (completed_at - started_at) is more of how long it took a CA to complete the intake from starting to submitting. If that's the desired metric, I think it makes sense to give it a different label.

File: app/services/asyncable_jobs_reporter.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-require "csv"
```
Comment (leikkisa) [inline app/services/asyncable_jobs_reporter.rb]: Should this still be in there since Reporter isn't included?

Comment (pkarman) [reply app/services/asyncable_jobs_reporter.rb]: yep csv lib is loaded by Rails by default, so it was superfluous to begin with.

**Comment by pkarman** [comment]
Thanks @leikkisa this is ready for a review now.

File: app/services/concerns/reporter.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Mixin methods for *Reporter classes. Statistics, dates, etc.
+
+require "csv"
```
Comment (nanotone) [inline app/services/concerns/reporter.rb]: Probably don't need this, right?

File: app/services/concerns/reporter.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    len = times.length
+    (times[(len - 1) / 2] + times[len / 2]) / 2.0
+  end
+
+  def average(times)
+    return 0 if times.empty?
+
+    times.sum.to_f / times.length
+  end
+end
```
Comment (nanotone) [inline app/services/concerns/reporter.rb]: This only nets a tiny bit of DRYness, but what about a little `percent` method, which I think ClaimReviewAsyncStatsReporter, RequestIssueReporter, and probably some future reporters, could use?

File: app/services/concerns/reporter.rb [hunk 5]
```diff
@@ -23,4 +21,8 @@ def average(times)
 
     times.sum.to_f / times.length
   end
+
+  def percent(nominator, denominator)
+    ((nominator / denominator.to_f) * 100).round(2)
+  end
```
Comment (nanotone) [inline app/services/concerns/reporter.rb]: Ooh nice ✨

**Pull request merged**